packages/react-native/src/NotifeeApiModule.ts

```diff
@@ -535,24 +536,28 @@ export default class NotifeeApiModule extends NotifeeNativeModule implements Mod
     return this.native.getNotificationCategories();
   };
 
-  public getNotificationSettings = (): Promise<IOSNotificationSettings> => {
+  public getNotificationSettings = (): Promise<NotificationSettings> => {
     if (isAndroid) {
-      // Android doesn't support this, so instead we
-      // return a dummy response to allow the permissions
-      // flow work the same on both iOS & Android
-      return Promise.resolve({
-        alert: 1,
-        badge: 1,
-        criticalAlert: 1,
-        showPreviews: 1,
-        sound: 1,
-        carPlay: 1,
-        lockScreen: 1,
-        announcement: 1,
-        notificationCenter: 1,
-        inAppNotificationSettings: 1,
-        authorizationStatus: 1,
-      } as IOSNotificationSettings);
+      return this.native
+        .getNotificationSettings()
+        .then(({ authorizationStatus }: Pick<NotificationSettings, 'authorizationStatus'>) => {
+          return {
+            authorizationStatus,
+            iOSSettings: {
```

Comment:
It seems strange that we are in an android conditional, and returning an iOSSettings object?

Reply:
it was to avoid platform dependent code 🤔 But, maybe it's not needed as we have `authorizationStatus`?

Reply:
Should it returns `undefined` instead of default values?